### PR TITLE
[INLONG-10873][SDK] Transform support factorial function

### LINKED ISSUE #10873: [Feature][SDK] Transform support factorial function
### Description

-parent issue #10796
Factorial(numeric)--returns the factorial of a non-negative
Example:
5! = 120

### Use case

_No response_

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by youzhi886**
Fixes #10873 
Motivation
Returns the factorial of a non-negative

### CODE REVIEW

File: inlong-sdk/transform-sdk/src/test/java/org/apache/inlong/sdk/transform/process/TestTransformArithmeticFunctionsProcessor.java [hunk 1]
```diff
@@ -174,6 +174,41 @@ public void testLog10Function() throws Exception {
...
     }
 
+    @Test
+    public void testFactorialFunction() throws Exception {
+        String transformSql = "select factorial(numeric1) from source";
+        TransformConfig config = new TransformConfig(transformSql);
+        TransformProcessor<String, String> processor = TransformProcessor
+                .create(config, SourceDecoderFactory.createCsvDecoder(csvSource),
+                        SinkEncoderFactory.createKvEncoder(kvSink));
+        // case1: 测试非负整数的阶乘
```
Comment (luchunliang) [inline inlong-sdk/transform-sdk/src/test/java/org/apache/inlong/sdk/transform/process/TestTransformArithmeticFunctionsProcessor.java:184]: Please use English as comment

Comment (youzhi886) [reply inlong-sdk/transform-sdk/src/test/java/org/apache/inlong/sdk/transform/process/TestTransformArithmeticFunctionsProcessor.java]: Thank you for the reminder. I have made the necessary changes.

Comment (youzhi886) [reply inlong-sdk/transform-sdk/src/test/java/org/apache/inlong/sdk/transform/process/TestTransformArithmeticFunctionsProcessor.java]: Thank you for the reminder. I have made the necessary changes.

File: inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/function/FactorialFunction.java [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+    private ValueParser numberParser;
+
+    /**
+     * Constructor
+     * 
+     * @param expr
+     */
+    public FactorialFunction(Function expr) {
+        numberParser = OperatorTools.buildParser(expr.getParameters().getExpressions().get(0));
```
Comment (luchunliang) [inline inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/function/FactorialFunction.java:44]: Please check expression list size at first.

**Comment by aloyszhang** [comment]
@youzhi886 Plz fix the code format and UT failures.

**Comment by aloyszhang** [comment]
please fix the conflicting file

**Comment by luchunliang** [comment]
Please use annotation to resolve conflicting file.
Please use independent UT class file to avoid UT conflicting file.

**Comment by luchunliang** [comment]
Please use annotation to resolve conflicting file.
Please use independent UT class file to avoid UT conflicting file.

**Comment by aloyszhang** [comment]
> Please use annotation to resolve conflicting file. Please use independent UT class file to avoid UT conflicting file.

@youzhi886 please fix these problems

**Comment by emptyOVO** [comment]
this pr will be continue complete in pr #11623

**Pull request closed without merging**